packages/eslint-plugin/src/rules/no-mixed-enums.ts

```diff
@@ -0,0 +1,222 @@
+import type { Scope } from '@typescript-eslint/scope-manager';
+import { DefinitionType } from '@typescript-eslint/scope-manager';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+
+import * as util from '../util';
+
+type AllowedType = ts.TypeFlags.Number | ts.TypeFlags.String;
```

Comment:
Should we switch to our own enum instead of (ab)using TS's enum?
It looks like you never return the checker value directly and instead manually specify the return type - so there's probs not much need to reuse their enum.